packages/engine-core/src/data-proxy/errors/utils/responseToError.ts

```diff
@@ -40,6 +53,24 @@ export async function responseToError(
   }
 
   if (response.status >= 500) {
+    let body
+    try {
+      body = await response.json()
+    } catch (err) {
+      throw new ServerError(info)
+    }
+
+    if (typeof body?.EngineNotStarted?.reason === 'string') {
+      throw new ServerError(info, body?.EngineNotStarted?.reason)
```

Comment:
```suggestion
      throw new ServerError(info, body.EngineNotStarted.reason)
```
Nit: should work with less `?` here